common/src/main/java/org/opensearch/ml/common/transport/connector/MLUpdateConnectorRequest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.transport.connector;
+
+import lombok.Builder;
+import lombok.Getter;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.core.common.io.stream.InputStreamStreamInput;
+import org.opensearch.core.common.io.stream.OutputStreamStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Map;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+@Getter
+public class MLUpdateConnectorRequest extends ActionRequest {
+    String connectorId;
+    Map<String, Object> updateContent;
+
+    @Builder
+    public MLUpdateConnectorRequest(String connectorId, Map<String, Object> updateContent) {
+        this.connectorId = connectorId;
+        this.updateContent = updateContent;
+    }
+
+    public MLUpdateConnectorRequest(StreamInput in) throws IOException {
+        super(in);
+        this.connectorId = in.readString();
+        this.updateContent = in.readMap();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(this.connectorId);
+        out.writeMap(this.getUpdateContent());
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException exception = null;
+
+        if (this.connectorId == null) {
+            exception = addValidationError("ML connector id can't be null", exception);
+        }
+
+        return exception;
+    }
+
+    public static MLUpdateConnectorRequest parse(XContentParser parser, String connectorId) throws IOException {
+        Map<String, Object> dataAsMap = null;
+        dataAsMap = parser.map();
```

Comment:
Do we accept any key/value for the new connector content? Should we do some validation ? For example checking if it's valid fiel in connector.

Reply:
Yes it supports add/update/delete any field in the OS document. If the value is null, the update request will remove the key. If the key is a new field, it will adds to the connector.

Reply:
agree, we should put the same connector validation here during the update.

Reply:
> If the key is a new field, it will adds to the connector.

This seems risky, user can hack by adding many new random fields

Reply:
The security concerns will be reviewed and addressed in the Pen tests.

Reply:
Let's add `TODO` in that case.